### ISSUE TITLE
Add inline text format

### ISSUE THREAD

**Issue by elong0527** [Merck/r2rtf#112]
We need design a function that allow user to define inline text format. In r2rtf, we already have `rtf_text` for the whole text. It might be a wrapper function to allow inline format like markdown part in `gt` https://gt.rstudio.com/articles/intro-creating-gt-tables.html#adding-parts-to-this-simple-table

A missing piece is to enable selected color in the text as we will need it for mock table. 

```
#' Enable color for mockup table
#'
#' The function will automatically replace text between `<` and `>` into blue.
#'
#' @param text a string.
#'
#' @export
rtf_mock_color <- function(text) {

  # Match text in glue expression { }
  match_text <- stringr::str_match_all(text, "\\{\\s*(.*?)\\s*\\}")
  match_text <- stats::na.omit(do.call(rbind, match_text))

  # Convert to blue color
  .x <- vapply(match_text[, 2], function(x) {

    # borrow from r2rtf:::rtf_text logic.
    x <- paste0("{\\cf26 <", x, ">}")
  }, FUN.VALUE = "character")
  .x <- .x[unique(names(.x))]

  # Apply the change
  res <- vapply(text, function(x) glue::glue_data(.x, x), FUN.VALUE = "character")
  names(res) <- names(text)
  res
}
```

**Comment by nanxstats** [comment]
Why not? It doesn't look like they have anything to do with metadata but all about RTF generation.

**Comment by BrianLang** [comment]
If we would like to put this into r2rtf, we can assign it to me and specify what needs to be done for it. 
It looks like `rtf_assemble()` is a function that already has an open issue #85 .

**Comment by elong0527** [comment]
Excellent, let's start from #85. 

For `rtf_mock_color`, we need to consider a more general strategy. (e.g. could we explore a markdown like syntax to provide inline format of text. 

I rename this issue to only focus on inline formatting.

**Comment by nanxstats** [comment]
For the inline markup, since this demo is already using `glue::glue()` internally, I'd suggest support evaluating glue expressions directly. Check out how {cli} does the inline markup:

- https://cli.r-lib.org/reference/inline-markup.html
- https://cli.r-lib.org/reference/themes.html

So that people can basically do this:

```r
rtf_rich_text(
  "This is {.emph important}.",
  theme = list(.emph = list(color = "blue0", `font-weight` = "bold"))
)
```

I think the `gt::md()` inline markup solution fulfills its own purpose, but it is limited to what the Markdown grammar offers: for example, no color styling. Technically you will need a CSS-like markup to support that, similar to what {cli} did.

**Comment by elong0527** [comment]
The syntax looks good to me.  @BrianLang would you be interested in exploring how `cli` achieve this behavior?

**Comment by BrianLang** [comment]
yeah this seems interesting

**Comment by BrianLang** [comment]
finally coming back to this.
```r
rtf_rich_text(
  "This is {.emph important}.",
  theme = list(.emph = list(color = "blue0", `font-weight` = "bold"))
)
```
something like this would then return a string of rtf code corresponding to a formatting of color change and bold?
`"This is \\colorblue0{\\fontbold{important}}"` (would need to look up rtf codes)

**Comment by elong0527** [comment]
Check the output from `r2rtf:::rtf_text` should be helpful https://merck.github.io/r2rtf/articles/rtf-text.html#text

Maybe `rtf_rich_text` is a wrapper of the `r2rtf:::rtf_text`. For example, the output of the function can be the executed code of 

```
glue::glue('This is {r2rtf:::rtf_text(important, color = "blue0", font_type = "b")')
```

**Comment by BrianLang** [comment]
thanks!

**Comment by elong0527** [comment]
closed by #184
